app/models/like.rb

```diff
@@ -0,0 +1,9 @@
+class Like < ApplicationRecord
+  belongs_to :post, inverse_of: :likes, foreign_key: :post_id, counter_cache: like_counter
+  belongs_to :user, inverse_of: :likes, foreign_key: :user_id, counter_cache: post_counter
+  # update likes counter
+
+  def update_like_counter
+    post.update(like_counter: post.likes.count)
+  end
```

Comment:
- [ ] You should increment the like counter for the post in this method and call the method after saving so when a user creates a like the counter should be incremented automatically. Please fix this issue.
  - You can consider the below code.
```ruby
after_save :update_like_counter
  def update_like_counter
    post.increment!(:like_counter)
  end